iocage_lib/ioc_json.py

```diff
@@ -964,11 +959,14 @@ def check_config(self, conf, default=False):
                 silent=self.silent,
                 exception=ioc_exceptions.CommandNeedsRoot)
 
-        if not default:
-            jail_conf = self.check_jail_config(conf)
-
         conf['CONFIG_VERSION'] = iocage_conf_version
 
+        # New style thin configuration jails won't have this. Only their
+        # defaults will
+        if thickconfig != 'THICK' and not default:
```

Comment:
@skarekrow @sonicaj I haven't removed my changes, yet, here. And maybe this is the wrong place in the code. But I think, when we change the behavior of `none`, we have to consider jails where `defaultrouter=none` has already been set by an iocage user explicitly. In these cases `defaultrouter=auto` of the "thick" default configuration would be overwritten by `defaultrouter=none` of the "thin" jail specific configuration.

Reply:
@joachimmathes you introduced a regression with this change for base jails. Please see the tests ;)

Reply:
Also we understand your perspective on this and it is absolutely right. We have opened up a new ticket for that which will improve how we keep track of props as to when it's changed if at all. For now, let's please stick to what @skarekrow suggested in the other PR because it is going to cause inconsistent behaviour with thin type configs.